iothub/device/src/InternalClient.cs

```diff
@@ -140,6 +141,7 @@ public InternalClient(IotHubConnectionString iotHubConnectionString, ITransportS
             pipelineContext.Set<ConnectionStatusChangesHandler>(OnConnectionStatusChanged);
             pipelineContext.Set<OnReceiveEventMessageCalledDelegate>(OnReceiveEventMessageCalled);
             pipelineContext.Set(_productInfo);
+            pipelineContext.Set(_pnpOptions);
```

Comment:
The only information available to the transport layer are the transport settings, connection string, method/twin/c2d callbacks, and the pipeline context. The `PnPOptions` type, containing the modelId is added to the pipeline context.